src/machines/videoPlayer.ts

```diff
@@ -1,220 +1,220 @@
+        loading: {
+          invoke: {
+            src: 'songLoader',
+            onDone: [
+              {
+                target: '#videoPalyerMachine.player.ready.hist',
+                actions: 'actionAssingSongData',
+              },
+            ],
+          },
+        },
+        finished: {
+          type: 'final',
+        },
+      },
+      onDone: {
+        target: '.loading',
+        description: 'cuando finished se ejecuta ( termina la canción )',
+        internal: false,
+      },
+      on: {
+        'DISLIKE': {
+          actions: ['actionDislikeSong', raise('SKIP')],
+        },
+        'AUDIO.TIME': {
+          actions: 'actionAssignTime',
+        },
+        'LIKE.TOGGLE': [
+          {
+            cond: 'guardIsLike',
+            actions: raise('UNLIKE'),
+          },
+          {
+            cond: 'guardIsUnlike',
+            actions: raise('LIKE'),
+          },
+        ],
+        'LIKE': {
+          actions: 'actionLikeSong',
+        },
+        'UNLIKE': {
+          actions: 'actionUnlikeSong',
+        },
+      },
     },
-    
-} ).withConfig( {
-    actions: {
-        actionAssingSongData: assign( {
-            artist: ( _, event: T_EventContext  ) => event.artist,
-            title: ( _, event: T_EventContext ) => event.title,
-            duration: ( _, event: T_EventContext ) => event.duration,
-            elapsed: 0,
-            likeStatus: "unliked",
-        } ),
-        actionUnlikeSong: assign( {
-            likeStatus: "unliked",
-        } ),
-        actionLikeSong: assign( {
-            likeStatus: "liked",
-        } ),
-        actionDislikeSong: assign( {
-            likeStatus: "disliked",
-        } ),
-        actionAssignTime: assign( {
-            elapsed: ( _, event: T_EventContext ) => event.elapsed,
-        } ),
-        actionEntryPlayAudio: () => {
-            console.log( "actionEntryPlayAudio" );
+    volume: {
+      initial: 'unmuted',
+      states: {
+        unmuted: {
+          on: {
+            'VOLUME.TOGGLE': {
+              target: 'muted',
+            },
+          },
         },
-        actionExitPlayAudio: () => {
-            console.log( "actionExitPlayAudio" );
+        muted: {
+          on: {
+            'VOLUME.TOGGLE': {
+              target: 'unmuted',
+            },
+          },
+        },
+      },
+      on: {
+        VOLUME: {
+          cond: 'guardVolumeWithinRange',
+          actions: 'actionAssignVolume',
         },
-        actionAssignVolume: assign( {
-            volume: ( _, event: T_EventContext ) => event.volume,
-        } ),
+      },
     },
-    /* guards: {
-    }, */
-} );
-//type VolumeEvent = { type: 'VOLUME', volume: number };
-
-type T_EventContext = {
-        title: string;
-        artist: string;
-        duration: number;
-        elapsed: number;
-        likeStatus: string;
-        volume: number;
-};
-
+  },
+}).withConfig({
+  services: {
+    songLoader: invokeSongLoader,
```

Comment:
Moved the call to `invokeSongLoader` to the services config to help typegen create the types